src/bin/cybergrape/main.rs

```diff
@@ -39,21 +41,30 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     let args = GrapeArgs::parse();
 
     // logic to parse commandline arguments for serial vs binaural
-    let SAMP_RATE: f32 = args.samp_rate;
-    let UPDATE_RATE: f32 = args.update_rate;
+    let update_rate: f32 = args.update_rate;
 
     let cmd = args.command;
+
+    let num_files;
+    let _outfile;
+    let _infile_samples;
+    let infile_gains;
+    let infile_ranges;
+
     match cmd {
-        Binaural(BinauralCommand) => {
-            let NUM_FILES: u32 = BinauralCommand.num_files;
-            let OUTFILE: String = BinauralCommand.outfile;
-            let INFILE_SAMPLES: Vec<Vec<f32>> = hound_reader(BinauralCommand.filenames);
-            let INFILE_GAINS: Vec<f32> = BinauralCommand.gains;
-            let INFILE_RANGES: Vec<f32> = BinauralCommand.ranges;
+        Binaural(binaural_command) => {
+            num_files = Some(binaural_command.num_files);
+            _outfile = Some(binaural_command.outfile);
+            _infile_samples = Some(hound_reader(binaural_command.filenames));
+            infile_gains = Some(binaural_command.gains);
+            infile_ranges = Some(binaural_command.ranges);
         }
 
-        Serial(SerialCommand) => {
-            let OUTFILE: String = SerialCommand.outfile;
+        Serial(serial_command) => {
+            num_files = Some(2);
+            _outfile = Some(serial_command.outfile);
+            infile_gains = Some(vec![1.0, 1.0]);
+            infile_ranges = Some(vec![1.0, 1.0]);
         }
     };
```

Comment:
```rust
    let (num_files, outfile, audio_settings) = match cmd {
        Binaural(binaural_command) => (
            binaural_command.num_files,
            binaural_command.outfile,
            Some((
                hound_reader(binaural_command.filenames),
                binaural_command.gains,
                binaural_command.ranges,
            )),
        ),
        Serial(serial_command) => (
            // TODO: add a num_tags/num_files argument to the serial command
            2,
            // outfile is common to both commands, though it means slightly different things
            serial_command.outfile,
            // the serial command doesn't have any audio samples and doesn't need gain/range info
            None,
        ),
    };
```
Alright if I'm understanding the serial command right, it will _also_ need to know the number of tags (aka the number of input files) you want to use, so we should add that as a parameter to the serial command. Then what remains is the input files and the ranges/gains... I suggest we wrap those in a single option to make it easier to match on later (to decide whether to output to a serial file or to binauralize)

After this we connect to the cube, record positional data for however long, store it all in a data structure then:
- if we have `audio_settings`, binauralize and output to `outfile`
- else, serialize positional data and output to `outfile`

Reply:
Ah, and because gains/ranges are now wrapped in an option, we can later default to 1.0 and 1.0 when we need to pass them to the sphericalizer

Reply:
That is an awesome idea. That totally works.